platform/solr/solr-appender/src/main/java/org/codice/solr/appender/SolrAppender.java

```diff
@@ -0,0 +1,126 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.solr.appender;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolrAppender implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrAppender.class);
+
+    private PersistentStore persistentStore;
+
+    private static final int DEFAULT_BATCH_SIZE = 100;
+
+    private static final int DEFAULT_PERIOD_IN_SECONDS = 10;
+
+    private int batchSize = DEFAULT_BATCH_SIZE;
+
+    private int period = DEFAULT_PERIOD_IN_SECONDS;
+
+    private ConcurrentLinkedQueue<PersistentItem> queue = new ConcurrentLinkedQueue<>();
+
+    private ScheduledExecutorService executorService;
+
+    private ScheduledFuture scheduledFuture;
+
+    public SolrAppender(PersistentStore persistentStore, ScheduledExecutorService executorService) {
+        this.persistentStore = persistentStore;
+        this.executorService = executorService;
+    }
+
+    @Override
+    public void handleEvent(Event event) throws IllegalArgumentException {
+        LOGGER.debug("Received decanter collection event on topic {}", event.getTopic());
+
+        PersistentItem item = new PersistentItem();
+        if (event.getProperty("id") == null) {
+            item.addIdProperty(UUID.randomUUID()
+                    .toString());
+        }
+
+        for (String key : event.getPropertyNames()) {
+            item.addProperty(key,
+                    event.getProperty(key)
+                            .toString());
+        }
+        queue.add(item);
+        if (queue.size() > batchSize) {
+            flushItems();
+        }
+
+    }
+
+    public void init() {
+        scheduledFuture = executorService.scheduleAtFixedRate(this::flushItems,
+                5,
+                period,
+                TimeUnit.SECONDS);
+    }
+
+    public void destroy() {
+        executorService.shutdown();
+    }
+
+    synchronized void flushItems() {
+        if (queue.isEmpty()) {
+            return;
+        }
+
+        try {
+            Collection<Map<String, Object>> items = new ArrayList<>();
+            items.addAll(queue);
+            queue.removeAll(items);
```

Comment:
❓ Should we wait until we know that the items have been successfully persisted before emptying the queue?